018-custom-id-support.md

```diff
@@ -0,0 +1,122 @@
+
+## Design proposal (Step 2)
+
+[design-proposal]: #design-proposal
+
+<!--
+This is the technical portion of the RFC. Explain the design in sufficient detail keeping in mind the following:
+
+- Its interaction with other parts of the system is clear
+- It is reasonably clear how the contribution would be implemented
+- Dependencies on libraries, tools, projects or work that isn't yet complete
+- New API routes that need to be created or modifications to the existing routes (if needed)
+- Any breaking changes and ways in which we can ensure backward compatibility.
+- Use Cases
+- Goals
+- Deliverables
+- Changes to documentation
+- Ways to scale the solution
+
+Ensure that you include examples, code-snippets etc. to allow the community to understand the proposed solution. **It would be best if the examples use naming conventions that you intend to use during the actual implementation so that changes can be suggested early on during the development.**
+
+Write your answer below.
+
+-->
+Resources that can have Custom IDs
+- Projects
+- Collections
+- Documents
+- Buckets
+- Files
+- Users
+- Function tags
+- Tasks
+- Webhooks
+
+The create resource endpoints for above resources will have a first required **id** parameter where users can send their custom id following the guidelines for validation. In every endpont, if the **id** parameter is received empty (required for backward compatibility with existing SDKs) or pre defined string, Appwrite will still generate the ids as it is already doing.
+
+We can use the existing key validator to validate the custom id as well.
+
+### Some guidelines for custom ids
+
+- Can be at most 32 characters
+- Can be Alphanumeric
+- Can contain `_` but not at the begining of the id
+- Cannot contain any other special characters
+
+### Prior art
+
+[prior-art]: #prior-art
+
+<!--
+
+Discuss prior art, both the good and the bad, in relation to this proposal. A
+few examples of what this can include are:
+
+- Does this functionality exist in other software and what experience has their
+  community had?
+- For other teams: What lessons can we learn from what other communities have
+  done here?
+- Papers: Are there any published papers or great posts that discuss this? If
+  you have some relevant papers to refer to, this can serve as a more detailed
+  theoretical background.
+
+This section is intended to encourage you as an author to think about the
+lessons from other software, provide readers of your RFC with a fuller picture.
+If there is no prior art, that is fine - your ideas are interesting to us
+whether they are brand new or if it is an adaptation from other software.
+
+Write your answer below.
+-->
+
+### Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+<!-- What parts of the design do you expect to resolve through the RFC process before this gets merged? -->
+Projects and Users being sensitive may be we can exclude them with custom ids.
```

Comment:
Not sure if we care about users ids being possible, the only problematic thing with a project is and ID called `console`.